peekingduck/training/src/metrics/tensorflow_metrics.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 AI Singapore
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from typing import Any, Dict, List, Optional, Union
+from omegaconf import DictConfig
+
+
+class TensorflowMetrics:
+    def get_metric(
+        self, metric_name: str, parameters: Union[DictConfig, dict] = {}
```

Comment:
Using dictionary default can cause behaviours mentioned in [here](https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python).

In general, mutable defaults can cause mutation downstream (was a first hand victim before).

Reply:
Does `parameters` need a default value? If not, it should be specified such that the caller always has to pass in something.